code/controllers/subsystem/shuttles.dm

```diff
@@ -205,6 +205,12 @@ SUBSYSTEM_DEF(shuttle)
 /datum/controller/subsystem/shuttle/proc/moveShuttle(shuttleId, dockId, timed, mob/user)
 	var/obj/docking_port/mobile/M = getShuttle(shuttleId)
 	var/obj/docking_port/stationary/D = getDock(dockId)
+	//check if the shuttle is on lockdown
+	if(M.lockeddown)
+		return 3
+	if(M.uses_lockdown)
+		M.lockeddown = TRUE
+		addtimer(VARSET_CALLBACK(M, lockeddown, FALSE), 15 SECONDS)
```

Comment:
is there any reason why you use `15 SECONDS` instead of the transit time of the shuttle?

Reply:
So the person has time to get off the shuttle before shuttle gets called back. 15 seconds seemed like a reasonable value in testing to get off the shuttle, even if not buckled in.